docs/pages/kubernetes-access/discovery/aws.mdx

```diff
@@ -0,0 +1,232 @@
+---
+title: Teleport EKS Auto-Discovery (Preview)
+description: Auto-Discovery of EKS clusters in AWS cloud.
+---
+
+Similarly to [AKS Auto-Discovery](./azure.mdx), EKS Auto-Discovery can automatically 
+discover any EKS cluster and enroll it in Teleport if its tags match the 
+configured labels. 
+
+(!docs/pages/kubernetes-access/discovery/includes/step-description.mdx!)
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- An AWS account with permissions to create and attach IAM policies.
+- An AWS account with `system:masters` RBAC access to EKS clusters.
+- A host to run the Teleport Discovery and Kubernetes services.
+- One or more EKS clusters running. 
+
+## Known limitations
+
+Due to the authorization method that AWS has chosen for EKS clusters, it is not 
+possible for the Teleport process to configure the necessary permissions to forward 
+requests to EKS clusters.
+This limitation exists because the authorization method on EKS clusters requires that the IAM role -
+used for authentication - exists in the 
+[`aws-auth`](https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html) `ConfigMap`. 
+
+In this `ConfigMap`, IAM roles are mapped to Kubernetes RBAC users or groups that are then 
+used by Kubernetes RBAC to grant access permissions. If the mapping does not exist, it results in 
+unauthorized requests. Therefore, Teleport cannot edit the `ConfigMap`
+without first having access to the cluster.
+
+If Teleport is not running with the same IAM identity that creates all clusters, 
+please ensure you configure the required permissions as described in Step 2. 
+
+## Step 1/3. Set up AWS IAM credentials
+
+For Teleport to discover EKS clusters, the instance
+running the Discovery Service requires an IAM policy that allows Teleport to list and 
+describe EKS clusters:
+
+```js
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*"
+        }
+    ]
+}      
+```
+
+## Step 2/3. Configure EKS cluster authorization
+
+
+When the Kubernetes Service uses an IAM role that is different from the one that 
+creates the clusters, you need to edit the `aws-auth` `Configmap` and append the 
+mapping between the Teleport IAM role and the RBAC permissions on each of the 
+discovered clusters. 
+
+For simplicity, it is possible to map the Teleport IAM role onto an existing 
+Kubernetes RBAC group such as `system:masters`, but Teleport recommends setting 
+up a separate Kubernetes RBAC group with the minimum required permissions.
+
+<Tabs>
+  <TabItem label="Minimum RBAC permissions (recommended)" >
+
+Connect to your target cluster with your credentials and create the following 
+resources using `kubectl`.
+
+### ClusterRole
```

Comment:
Wouldn't Teleport create ClusterRole/Binding if aws-auth configmap allows it access?

Reply:
In the case of AWS, this does not happen. Although it is technically possible, the Kubernetes RBAC role configured in the configmap probably grants the required Teleport permissions instead of permissions to create `ClusterRoles` and `ClusterRoleBindings` for security reasons.

Should I create a small PR to create them if Teleport has permissions?

Reply:
> the configmap probably grants the required Teleport permissions

Maybe I'm missing something but if the configmap grants the required permissions, why do they need to create these role/binding?

> Should I create a small PR to create them if Teleport has permissions?

I think it would be worthwhile, yeah. The less users have to configure themselves the better. Worst case we'll just get an access denied error and fail gracefully.

Reply:
> Maybe I'm missing something but if the configmap grants the required permissions, why do they need to create these role/binding?

If it grants the required permissions then you don't need to do anything. The issue arises when the `aws-auth` configmap maps the Teleport IAM role to a Kubernetes RBAC group that allows creating ClusterRole and ClusterRoleBindings but does not allow impersonation.

Reply:
Right, I think the way this section structured is a little confusing, mostly because of the split between 2 options. I wonder if restructuring it like this would make more sense for a reader:

1. Say that Teleport needs X, Y and Z permissions in the cluster and there are following options to pick from:
  - If you don't have any roles with these permissions in the cluster, create this cluster role/binding.
  - If you already have a role in the cluster with these permissions, you can use it.
  - You can also use builtin system:masters which has all permissions but not recommended in production.
2. Map Teleport IAM role to the Kube role from ^^^ using `aws-auth`.

Would this flow be clearer you think? @tigrato